### DISCUSSION TITLE
how to stop notification auto show up?

### DISCUSSION THREAD

**Discussion by jsw-davidhuang** [firebase/flutterfire#7413] [Q&A]
as the title, i had add firebase messaging, but when the backend send out a notification, the phone auto pop up the notification(app is in background), but i want to process the data first, how can i make it? also i had also try ask the backend send out data instead of notification, still the same(app in background)

**Comment by nivisi** [comment]
+1 here, is that possible?

**Reply by jsw-davidhuang** [reply]
yes, i had solve it. it turn out the backend sent out in data field but forgot to delete the notification field/ when it only send out data field,it can process by app first then app send out notification.

**Reply by nivisi** [reply]
@jsw-davidhuang we also tried such an approach, but it doesn't work for us on iOS when the app is killed. Does it for you?
